rewrite-gradle/src/test/java/org/openrewrite/gradle/ChangeDependencyTest.java

```diff
@@ -799,4 +800,53 @@ void makeChangesInKotlinDependencyManagementImports() {
           )
         );
     }
+
+    @Test
+    void noDuplicateJacksonDatabindDependenciesInGradle() {
+        rewriteRun(
+          spec -> spec.beforeRecipe(withToolingApi())
+            .recipes(
+              new ChangeDependency(
+                "com.fasterxml.jackson.core",
+                "jackson-databind",
+                "tools.jackson.core",
+                null,
+                "3.0.x",
+                null,
+                null,
+                null
+              ),
+              new ChangeDependency(
+                "com.fasterxml.jackson.datatype",
+                "jackson-datatype-jsr310",
+                "tools.jackson.core",
+                "jackson-databind",
+                "3.0.x",
+                null,
+                null,
+                null
+              )
+            ),
+          buildGradle(
+            //language=gradle
+            """
+              plugins {
+                  id("java-library")
+              }
+              repositories {
+                  mavenCentral()
+              }
+              dependencies {
+                  implementation("com.fasterxml.jackson.core:jackson-databind:2.19.0")
+                  implementation("com.fasterxml.jackson.datatype:jackson-datatype-jsr310:2.19.0")
+              }
+              """,
+            spec -> spec.after(buildGradle ->
+              assertThat(buildGradle)
+                .containsOnlyOnce("tools.jackson.core:jackson-databind:3")
+                .doesNotContain("datatype")
```

Comment:
For context: without the above changes we saw `jackson-databind:3` duplicated twice, as both map to the new artifact & version.